### PR TITLE
[NAE-1156] View Deprecation and All Data

### PR DESCRIPTION
**Pull request by Kovy95**
# Description

Remove view tag.

Fixes [NAE-1156]

## Dependencies
none

### Third party dependencies
none

### Blocking Pull requests
none

## How Has Been This Tested?

manually

### Test Configuration

| Name                | Tested on |
|---------------------| --------- |
| OS                  |      Linux Mint     |
| Runtime             |     NodeJS      |
| Dependency Manager  |   NPM        |
| Framework version   |  Angular 13.3         |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes have been checked, personally or remotely, with @...
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes:
    - [x] Lint test
    - [x] Unit tests
    - [x] Integration tests
- [x] I have checked my contribution with code analysis tools:
    - [x] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [x] [Snyk](https://app.snyk.io/org/netgrif)
- [x] I have made corresponding changes to the documentation:
    - [x] Developer documentation
    - [x] User Guides
    - [x] Migration Guides

[NAE-1156]: https://netgrif.atlassian.net/browse/NAE-1156?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: projects/netgrif-components-core/src/lib/data-fields/models/abstract-data-field.ts [hunk 1]
```diff
@@ -323,6 +323,14 @@ export abstract class DataField<T> {
         return this._reverting;
     }
 
+    /**
+     * This function resolve type of component for HTML
+     * @returns type of component in string
+     */
+    public getComponentType(): string {
+        return !!this.component && !!this.component.name ? this.component.name : '';
```
Comment (minop) [inline projects/netgrif-components-core/src/lib/data-fields/models/abstract-data-field.ts:331]: `return this.component?.name ?? '';`

File: projects/netgrif-components-core/src/lib/task-content/services/field-converter.service.ts [hunk 2]
```diff
@@ -36,30 +36,14 @@ export class FieldConverterService {
...
-                        item.description, item.layout, item.validations, type);
-                }
-                if (item.component !== undefined && item.component.name !== undefined && (item.component.name === 'editor' ||
-                    item.component.name === TextFieldView.HTMLTEXTAREA || item.component.name === TextFieldView.RICHTEXTAREA)) {
-                    return new TextAreaField(item.stringId, item.name, this.resolveTextValue(item, item.value),
-                        item.behavior, item.placeholder, item.description, item.layout, item.validations, type, item.component,
-                        item.parentTaskId);
+                if (item.component !== undefined && item.component.name !== undefined && (item.component.name === 'textarea' ||
+                    item.component.name === 'richtextarea' || item.component.name === 'htmltextarea' ||
+                    item.component.name === 'editor' || item.component.name === 'htmlEditor' || item.component.name === 'area') ) {
```
Comment (minop) [inline projects/netgrif-components-core/src/lib/task-content/services/field-converter.service.ts:41]: this if is quite monstrous...

the component name can be safely accessed with `item.component?.name`

the 6 options that result in a TextAreaField should be held in some array or other collection and the condition should just check if the `item.component?.name` is in the collection or not.

File: projects/nae-example-app/src/app/doc/tabbed-case-view/tabbed-case-view/tabbed-case-view.component.scss [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 .case-view-search-container {
-    padding: 8px 16px;
+    padding: 8px 8px;
 }
 
 .case-header-padding {
-    padding-bottom: 16px;
-    padding-top: 16px;
+    padding-bottom: 8px;
+    padding-top: 8px;
```
Comment (minop) [inline projects/nae-example-app/src/app/doc/tabbed-case-view/tabbed-case-view/tabbed-case-view.component.scss:13]: why are these styles only changed in the example app?

**Review by minop — CHANGES_REQUESTED** [review]

File: projects/netgrif-components-core/src/lib/data-fields/models/abstract-data-field.ts (hunk 1 above)
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/data-fields/models/abstract-data-field.ts]: done

File: projects/netgrif-components-core/src/lib/task-content/services/field-converter.service.ts (hunk 2 above)
Comment (Kovy95) [reply projects/netgrif-components-core/src/lib/task-content/services/field-converter.service.ts]: done

File: projects/nae-example-app/src/app/doc/tabbed-case-view/tabbed-case-view/tabbed-case-view.component.scss (hunk 3 above)
Comment (Kovy95) [reply projects/nae-example-app/src/app/doc/tabbed-case-view/tabbed-case-view/tabbed-case-view.component.scss:13]: done

**Review by minop — APPROVED** [review]

**Review by minop — APPROVED** [review]

**Review by machacjozef — APPROVED** [review]

**Review by mladoniczky — APPROVED** [review]

**Pull request merged**